sympy/sets/fancysets.py

```diff
@@ -766,6 +766,15 @@ def _sup(self):
     def _boundary(self):
         return self
 
+    def as_relational(self, x):
+        """Rewrite a Range in terms of equalities and logic operators. """
+        from sympy.functions.elementary.integers import floor
+        i = (x - (self.inf if self.inf.is_finite else self.sup))/self.step
+        return And(
+            Eq(i, floor(i)),
```

Comment:
what about using `Q.integer(i)` instead of `floor(i) == i`?

Reply:
Any logic that is good at handling Relationals will break when encountering a Q. How would the Q.integer(i) evaluate?
```python
>>> Q.integer(1)
Q.integer(1)
>>> ask(_)
True
```